lib/chef/provider/git.rb

```diff
@@ -323,6 +319,12 @@ def git(*args)
         ["git", *args].compact.join(" ")
       end
 
+      def run_git_command(args, run_opts = {})
```

Comment:
so here's where you need to do `git_run_command(*args, **opts)` and then you can remove the arrays and hashes from the callers